src/components/SkillCreator/SkillCreator.js

```diff
@@ -33,95 +33,383 @@ import ChevronRight from 'material-ui/svg-icons/navigation/chevron-right';
+        codeChanged: false,
+        showDeleteBox: false,
+        groupValue: this.props.location.pathname.split('/')[1],
+        oldGroupValue: this.props.location.pathname.split('/')[1],
+        languageValue: this.props.location.pathname.split('/')[4],
+        oldLanguageValue: this.props.location.pathname.split('/')[4],
+        expertValue: this.props.location.pathname.split('/')[2],
+        oldExpertValue: this.props.location.pathname.split('/')[2],
+        commitId: this.props.location.pathname.split('/')[5],
+        groupSelect: false,
+        languageSelect: false,
+        expertSelect: false,
+        date: '',
+        author: '',
+        oldImageUrl: '',
+        imageUrl: '<image_url>',
+        image_name_changed: false,
+        fontSizeCode: 14,
+        editorTheme: 'github',
+        showAdmin: false,
+        deleteDisabled: true,
+        slideState: 1, // 1 means in middle, 2 means preview collapsed
+        colSkill: this.props.hasOwnProperty('revertingCommit') ? 12 : 8,
+        colPreview: this.props.hasOwnProperty('revertingCommit') ? 0 : 4,
+        prevButton: 0, // 0 means disappear, 1 means appear
+      };
+      let fonts = [14, 16, 18, 20, 24, 28, 32, 40];
+      let themes = [
+        'monokai',
+        'github',
+        'tomorrow',
+        'kuroir',
+        'twilight',
+        'xcode',
+        'textmate',
+        'solarized_dark',
+        'solarized_light',
+        'terminal',
+      ];
+      for (let i = 0; i < fonts.length; i++) {
+        fontsizes.push(
+          <MenuItem
+            value={fonts[i]}
+            key={fonts[i]}
+            primaryText={`${fonts[i]}`}
+          />,
+        );
+      }
+      for (let i = 0; i < themes.length; i++) {
+        codeEditorThemes.push(
+          <MenuItem
+            value={themes[i]}
+            key={themes[i]}
+            primaryText={`${themes[i]}`}
+          />,
+        );
+      }
+    }
   }
 
-  componentDidMount = () => {
-    document.title = 'SUSI.AI - Create Skill';
-    self = this;
-    self.loadgroups();
-    // send code to CodeView in botbuilder
-    if (this.props.botBuilder) {
-      this.setState({
-        slideState: 0,
-        colSkill: 12,
-        colPreview: 0,
-        prevButton: 0,
+  updateData = skillData => {
+    if (skillData.image) {
+      this.imgUrl = `${urls.API_URL}/cms/getImage.png?model=general&language=${
+        this.state.languageValue
+      }&group=${this.state.groupValue}&image=${skillData.image}`;
+    } else {
+      this.imgUrl = '/favicon-512x512.jpg';
+    }
+    this.setState({
+      image: this.imgUrl,
+    });
+  };
+
+  loadlanguages() {
+    if (languages.length === 0) {
+      $.ajax({
+        url: urls.API_URL + '/cms/getAllLanguages.json',
+        dataType: 'jsonp',
+        jsonp: 'callback',
+        crossDomain: true,
+        success: function(data) {
+          data = data.languagesArray;
+          this.setState({ languages: data });
```

Comment:
It will throw an error. Use fat arrow syntax to bind the function with the scope